### PR TITLE
Add thin logger as demanded by #47

### PR DESCRIPTION
**Pull request by danielwolff1**
# Implement thin logging functionality for `splinepy`

This is my first PR, hopefully I'm doing it right...

## Related Issues

* [Create own logger class for `splinepy`](https://github.com/tataratat/splinepy/issues/47)

## Description

* Moved `configure()` from `splinepy.utils` to `splinepy.log`
* Added `debug()`, `info()` and `warning()` methods in `splinepy.log`
* Added a base class `SplinepyBase` implemented in `_base.py`, which is now the base class for all `splinepy` classes and so far only implements logging functions `logd()`, `logi()` and `logw()` utilizing the functionality provided by `splinepy.log` 
* Changed logging in all other classes to use the functions provided by `SplinepyBase` instead of the native `logging` module

### CODE REVIEW

File: splinepy/_spline.py [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-import logging
 import copy
 import os
-import itertools
 
 import numpy as np
 
 from splinepy import utils
 from splinepy import io
-from splinepy import _splinepy as core
```
Comment (j042) [inline splinepy/_spline.py:11]: What happened to this line? does this work?

File: splinepy/_spline.py [hunk 2]
```diff
@@ -217,7 +214,7 @@ def whatami(self, how_dare_you):
         --------
         None
         """
-        logging.warning("Spline - Excuse me, you cannot tell me what I am.")
+        self.logw("Excuse me, you cannot tell me what I am.")
```
Comment (j042) [inline splinepy/_spline.py:217]:
```suggestion
        self._logw("Excuse me, you cannot tell me what I am.")
```
missing `_`?

File: splinepy/_spline.py (hunk 1 above)
Comment (danielwolff1) [reply splinepy/_spline.py:11]: whopsie 🙈

File: splinepy/_spline.py (hunk 2 above)
Comment (danielwolff1) [reply splinepy/_spline.py]: Yeah, my bad, thanks for pointing that out! 👍

**Review by j042 — APPROVED** [review]
good stuff

**Pull request merged**